[READY] Gives Stacking Machines and their consoles machine boards

iskyp opened the pull request:
:cl: iksyp
add: stacking machines and their consoles no longer dissapear into the shadow realm when broken
add: you can link the stacking machine and its console by using a multitool
/:cl:
TODO:
- [x] Make the console and machine linkable.
- [x] Break up the engineering nodes, since I'm a liar.
- [x] Change the recipes to be a little less shit.

Commit:
patch-2

ExcessiveUseOfCobblestone commented on code/modules/research/techweb/all_nodes.dm:
```diff
@@ -93,7 +93,7 @@
 	prereq_ids = list("base")
 	design_ids = list("solarcontrol", "recharger", "powermonitor", "rped", "pacman", "adv_capacitor", "adv_scanning", "emitter", "high_cell", "adv_matter_bin",
-	"atmosalerts", "atmos_control", "recycler", "autolathe", "high_micro_laser", "nano_mani", "mesons", "thermomachine", "rad_collector", "tesla_coil", "grounding_rod", "apc_control", "cell_charger")
+	"atmosalerts", "atmos_control", "recycler", "autolathe", "high_micro_laser", "nano_mani", "mesons", "thermomachine", "rad_collector", "tesla_coil", "grounding_rod", "apc_control", "cell_charger", "stack_console", "stack_machine")
```
We need to break this up eventually

iskyp replied:
eh, probably, but i'm not gonna do it.

iskyp replied:
i lied

PKPenguin321 commented:
can you link them if you rebuild them?

iskyp commented:
i'm not actually sure, i'll test it when i get home

Jalleo commented:
No. This is map defined stuff I believe.

ExcessiveUseOfCobblestone commented:
just have the computer and machine key editable with a multitool or something.

Fox-McCloud commented on code/modules/mining/machine_stacking.dm:
```diff
@@ -7,6 +7,7 @@
 	desc = "Controls a stacking machine... in theory."
 	density = FALSE
 	anchored = TRUE
+	circuit = /obj/item/circuitboard/machine/stacking_unit_console
```
You mind fixing the problem with this? below `speed_process` is set to `TRUE`. This machine doesn't need it; it's just a console.

That said, `/obj/machinery/mineral/stacking_machine` hypothetically does (though I'm not sure, anymore, what with `HasProximity`

Fox-McCloud commented:
Nice, but you're going to cause confusion. The console attempts to locate the stacking machine when it's created; if it can't find one, it immediately deletes itself.

Probably going to be a lot of people going "huh, uh, didn't I build this?"

improvedname commented:
Just remove the stacking machine old people the future is here O.R.M

Fox-McCloud commented:
You're confusing the furnace with the stacking machine, I think.

The stacking machine is still used in disposals. It takes in stacks and combined them together inside the stacker, then you can release them with a console. They're still very useful; having one inside cargo could effectively mean that you could have material sheets automatically pulled out of cargo-sorting for you, and you could pretty much leave the recycling windoor open so as soon as the recycling machine released its stacks, it would go back to cargo without fear of it passing through the recycler again.

Commit:
does everything on the TODO list

FUCK YEAH
*breaks up the engineering node design ID's for readability
*linking the device via multitool is possible
*The machinery construction "recipes" are less shit and make a bit more sense

iskyp commented:
everything is done woohoo
i didn't even need to use the DNM label but i did i guess even though i fixed it in like 2 hours

Bawhoppen commented:
Finally. One step closer to completely rebuildable station. :+1:x:100:

Fox-McCloud commented on code/modules/mining/machine_stacking.dm:
```diff
@@ -35,6 +33,13 @@
 	user << browse(dat, "window=console_stacking_machine")
 
+/obj/machinery/mineral/stacking_unit_console/multitool_act(mob/living/user, obj/item/I)
+	var/obj/item/multitool/M = I
```
You usually want to declare vars like this *after* doing a typecheck.

iskyp replied:
I tried, but it broke when I did it that way. as in, it stopped working entirely

SpaceManiac replied:
Don't know why, it's working fine in the other method.

iskyp replied:
fucking byond, it works fine now, but last time it was throwing compiler errors at me.
next commit should fix all issues

Fox-McCloud commented on code/game/objects/items/circuitboards/machine_circuitboards.dm:
```diff
@@ -864,3 +864,17 @@
+	build_path = /obj/machinery/mineral/stacking_unit_console
+	req_components = list(
+		/obj/item/stack/sheet/glass = 2,
+		/obj/item/stock_parts/manipulator = 2)
```
This is pretty much just a computer; the stacker does all the work. Why not just glass and cable coil?

iskyp replied:
why didnt I think of that hold on

Commit:
coil

improvedname commented:
@fox-mccloud last time i used stacker was at gulag couldnt even drop the minerals i mined cuz it only drops stacks

SpaceManiac commented on code/modules/mining/machine_stacking.dm:
```diff
@@ -79,6 +85,18 @@
+	if(istype(I, /obj/item/multitool))
+		var/obj/item/multitool/M = I
+		if(!istype(M.buffer, /obj/machinery/mineral/stacking_unit_console))
+			to_chat(user, "<span class='warning'>The [I] has no linkage data in it's buffer.</span>")
```
`it's` -> `its`

SpaceManiac reviewed (changes requested):
I think you may need handling for when `machine` or `CONSOLE` is null, so using unlinked machines doesn't just runtime everywhere.

Commit:
did somebody say RUNTIMES?

*fixed runtimes when there is no link on the stacking machine console
*added feedback for unlinked consoles
*it's -> its
*proper typecheck

Commit:
Merge branch 'master' into patch-2

SpaceManiac commented on code/modules/mining/machine_stacking.dm:
```diff
@@ -19,6 +19,11 @@
 	. = ..()
+
+	if(!machine)
+		to_chat(user, "<span class='notice'>The [src] is not linked to a machine!</span>")
```
`[src]` adds "The" automatically. As is, it will come out "The the stacking machine console".

Commit:
achieved with [src]
